### PR TITLE
Print `val` for class parameters

### PR DESCRIPTION
**Pull request by nicolasstucki**

### CODE REVIEW

File: library/src/scala/tasty/util/ShowSourceCode.scala [hunk 1]
```diff
@@ -621,6 +621,17 @@ class ShowSourceCode[T <: Tasty with Singleton](tasty0: T) extends Show[T](tasty
...
+        case DefDef("<init>", _, _, _, _) =>
+          val ClassDef(_, _, _, _, body) = arg.owner.owner
+          body.collectFirst {
+            case vdef @ ValDef(`name`, _, _) if vdef.flags.isParamAccessor =>
+              if (!vdef.flags.isLocal && !vdef.flags.isCaseAcessor)
+                this += "val " // TODO `var`s
+          }
+        case _ =>
+      }
+
```
Comment (liufengyun) [inline library/src/scala/tasty/util/ShowSourceCode.scala]: Rename `printArgDef` to `printParam`? My impression is that `param` is used at definition site, `arg` is used usually used at calling site.

File: library/src/scala/tasty/util/ShowSourceCode.scala [hunk 2]
```diff
@@ -621,6 +621,17 @@ class ShowSourceCode[T <: Tasty with Singleton](tasty0: T) extends Show[T](tasty
 
     def printArgDef(arg: ValDef): Unit = {
       val ValDef(name, tpt, rhs) = arg
+      arg.owner match {
+        case DefDef("<init>", _, _, _, _) =>
+          val ClassDef(_, _, _, _, body) = arg.owner.owner
+          body.collectFirst {
+            case vdef @ ValDef(`name`, _, _) if vdef.flags.isParamAccessor =>
+              if (!vdef.flags.isLocal && !vdef.flags.isCaseAcessor)
+                this += "val " // TODO `var`s
```
Comment (liufengyun) [inline library/src/scala/tasty/util/ShowSourceCode.scala]: More todos: access modifiers, default parameters, implicits.

**Comment by nicolasstucki** [comment]
@liufengyun could you review the second commit

File: library/src/scala/tasty/util/ShowSourceCode.scala [hunk 3]
```diff
@@ -1053,6 +1059,48 @@ class ShowSourceCode[T <: Tasty with Singleton](tasty0: T) extends Show[T](tasty
...
+            printWithin(within)
+            this += "] "
+            printedPrefix = true
+          case _ =>
+        }
+      }
+      printedPrefix
+    }
+
+    def printFullClassName(tp: TypeOrBounds): Unit = {
```
Comment (liufengyun) [inline library/src/scala/tasty/util/ShowSourceCode.scala]: `TypeOrBounds` -> `Type`? The same for the next line.

File: library/src/scala/tasty/util/ShowSourceCode.scala [hunk 4]
```diff
@@ -619,15 +623,24 @@ class ShowSourceCode[T <: Tasty with Singleton](tasty0: T) extends Show[T](tasty
...
+              if (!vdef.flags.isLocal) {
+                var printedPrefix = false
+                if (vdef.flags.isOverride) {
+                  this += "override "
+                  printedPrefix = true
+                }
+                printedPrefix  |= printProtectedOrPrivate(vdef)
+                if (vdef.flags.isMutable) this += "var "
+                else if (!printedPrefix && vdef.flags.isCaseAcessor) this // val not explicitly needed
+                else this += "val "
```
Comment (liufengyun) [inline library/src/scala/tasty/util/ShowSourceCode.scala]: I would negate the second `else if` and put `this += "val "` as the body.

File: library/src/scala/tasty/util/ShowSourceCode.scala [hunk 5]
```diff
@@ -619,15 +623,24 @@ class ShowSourceCode[T <: Tasty with Singleton](tasty0: T) extends Show[T](tasty
...
           val ClassDef(_, _, _, _, body) = arg.owner.owner
           body.collectFirst {
             case vdef @ ValDef(`name`, _, _) if vdef.flags.isParamAccessor =>
-              if (!vdef.flags.isLocal && !vdef.flags.isCaseAcessor)
-                this += "val " // TODO `var`s
+              if (!vdef.flags.isLocal) {
+                var printedPrefix = false
+                if (vdef.flags.isOverride) {
+                  this += "override "
+                  printedPrefix = true
```
Comment (liufengyun) [inline library/src/scala/tasty/util/ShowSourceCode.scala]: Why skip `val ` (line 641) if we have `override`?

File: library/src/scala/tasty/util/ShowSourceCode.scala (hunk 4 above)
Comment (liufengyun) [reply library/src/scala/tasty/util/ShowSourceCode.scala]: The name `printedPrefix` is not informative enough to understand the logic here.

File: library/src/scala/tasty/util/ShowSourceCode.scala (hunk 5 above)
Comment (nicolasstucki) [reply library/src/scala/tasty/util/ShowSourceCode.scala]: We skip `val` if we have NOT `override`

**Pull request merged**